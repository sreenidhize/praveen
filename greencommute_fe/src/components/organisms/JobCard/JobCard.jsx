import React from "react";
import PropTypes from "prop-types";
import { Grid, Box } from "@material-ui/core";
import CompanyLogo from "../../atom/CompanyLogo/CompanyLogo";
import JobCardLabel from "../../atom/JobCardLabel/JobCardLabel";
import CommuteIcons from "../../molecules/CommuteIcons/CommuteIcons";
import { makeStyles } from "@material-ui/core";
import MoreHorizOutlinedIcon from "@material-ui/icons/MoreHorizOutlined";

const useStyles = makeStyles({
  iconsCardLarge: {
    marginTop: "10px",
  },

  boxStyle: {
    display: "flex",
    alignItems: "flex-end",
    justifyContent: "flex-end",
    minWidth: "3rem",
  },
  iconsCardSmall: {
    display: "flex",
    alignItems: "flex-end",
    justifyContent: "flex-end",
    marginLeft:"23px"
  },

  dateInfo: {
    position: "absolute",
    right: "38px",
    display: "flex",
  },

  largeCard: {
    height: "308px",
    padding: "25px",
    border: "solid 2px #e3f3f6",
    borderRadius: "4px",
    position: "relative",
    background: "white",
    cursor: "pointer",
  },

  smallCard: {
    width: "100%",
    height: "138px",
    padding: "30px",
    border: "solid 2px #e3f3f6",
    borderRadius: "10px",
    position: "relative",
    background: "white",
    cursor:"pointer",
  },

  cardHilight:{
    border: "solid 2px #5AC568"
  },

  moreIcon: {
    color: "#5f7381",
    margin: "-3px",
    paddingLeft: "24px",
  },

  smallCardInfo: {
    marginLeft: "15px",
    width: "420px",
  },
});
const JobCard = ({
  isGrid,
  source,
  alt,
  jobTitle,
  companyName,
  location,
  commuteIcons,
  isHighlight,
  ...rest
}) => {
  const classes = useStyles();
  const MainIconsStyle = isGrid ? "" : classes.iconsCardSmall;
  const IconsStyle = isGrid ? classes.iconsCardLarge : "";

  const getClassName = () => {
    if(isGrid && isHighlight) {
      return `${classes.largeCard} ${classes.cardHilight}`
    }
    if(isGrid === false && isHighlight){
      return `${classes.smallCard} ${classes.cardHilight}`
    }
    if(isGrid){
      return `${classes.largeCard}`
    }
    return `${classes.smallCard}`
  }

  return (
    <Grid
      container
      direction={isGrid ? "column" : "row"}
      className={getClassName()}
      data-testid="jobCard"
      {...rest}
    >
      <div className={classes.dateInfo}>
        <JobCardLabel label="2d" />
        <MoreHorizOutlinedIcon className={classes.moreIcon} />
      </div>
      <Grid item>
        <CompanyLogo source={source} alt={alt} isGrid={isGrid}></CompanyLogo>
      </Grid>
      <Grid item style={{width:"81%"}}>
        <Grid
          item
          container
          direction="column"
          justify="space-between"
          className={` ${isGrid ? "" : classes.smallCardInfo}`}
        >
          <Grid item>
            <JobCardLabel label={jobTitle} style={true} />
          </Grid>
          <Grid item>
            <JobCardLabel label={companyName} />
          </Grid>
          <Grid item>
            <JobCardLabel label={location} />
          </Grid>
        </Grid>
      </Grid>

      <Grid item className={MainIconsStyle}>
        {isGrid ? (
          <div className={IconsStyle}>
            <CommuteIcons icons={commuteIcons} isTextShown={isGrid} />
          </div>
        ) : (
          <Box className={classes.boxStyle} width="6.9rem">
            <CommuteIcons icons={commuteIcons} isTextShown={isGrid} />
          </Box>
        )}
      </Grid>
    </Grid>
  );
};

JobCard.propTypes = {
  isGrid: PropTypes.bool,
  source: PropTypes.string,
  alt: PropTypes.string,
  jobTitle: PropTypes.string,
  companyName: PropTypes.string,
  location: PropTypes.string,
  commuteIcons: PropTypes.array,
};

export default JobCard;
