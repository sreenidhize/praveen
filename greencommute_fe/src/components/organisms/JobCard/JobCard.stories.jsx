import React from "react";
import JobCard from "./JobCard";
import imageFile from "../../../assets/hp.png";
export default {
  title: "Organisms/Card",
  component: JobCard,
};
const JobCardTemplate = (args) => <JobCard {...args} />;

export const Primary = JobCardTemplate.bind({});

Primary.args = {
  isGrid: true,
  source: imageFile,
  alt: "HP logo",
  jobTitle: "User Experience Designer",
  companyName: "HP",
  location: "Hitech City, Telangana",
  commuteIcons: [
    { iconName: "bus" },
    { iconName: "metro" },
    { iconName: "car" },
    { iconName: "bike" },
  ],
};

export const Secondary = JobCardTemplate.bind({});

Secondary.args = {
  isGrid: false,
  source: imageFile,
  alt: "HP logo",
  jobTitle: "User Experience Designer",
  companyName: "HP",
  location: "Hitech City, Telangana",
  commuteIcons: [
    { iconName: "bus" },
    { iconName: "metro" },
    { iconName: "car" },
    { iconName: "bike" },
  ],
};

