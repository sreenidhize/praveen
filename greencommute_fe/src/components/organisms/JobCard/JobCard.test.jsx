import React from "react";
import JobCard from "./JobCard";
import { render } from "@testing-library/react";
const icons = [
  { iconName: "bus" },
  { iconName: "metro" },
  { iconName: "car" },
  { iconName: "bike" },
];
describe("JobCard renders correctly", () => {
  test("should match snapshot", () => {
    const { asFragment } = render(<JobCard commuteIcons={icons} />);
    expect(asFragment()).toMatchSnapshot();
  });

  test("JobCard should not be empty", () => {
    const { container } = render(
      <JobCard
        isGrid={true}
        source="https://upload.wikimedia.org/wikipedia/commons/4/44/BMW.svg"
        alt="HP logo"
        jobTitle="User Experience Designer"
        companyName="BMW"
        location="Hi-Tech City, Telangana"
        commuteIcons={icons}
      />
    );
    expect(container.getElementsByTagName("img").length).toBe(1);
    expect(container.getElementsByTagName("h4").length).toBe(5);
  });
});
