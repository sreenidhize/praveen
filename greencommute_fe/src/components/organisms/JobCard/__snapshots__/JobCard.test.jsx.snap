// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JobCard renders correctly should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root makeStyles-smallCard-6 MuiGrid-container"
    data-testid="jobCard"
  >
    <div
      class="makeStyles-dateInfo-4"
    >
      <h4
        class="MuiTypography-root makeStyles-secondary-11 MuiTypography-h4"
      >
        2d
      </h4>
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root makeStyles-moreIcon-8"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M6 10c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm12 0c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm-6 0c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
        />
      </svg>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item"
    >
      <div
        class="MuiAvatar-root MuiAvatar-circle makeStyles-secondary-13 MuiAvatar-colorDefault"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiAvatar-fallback"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
          />
        </svg>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item"
      style="width: 81%;"
    >
      <div
        class="MuiGrid-root  makeStyles-smallCardInfo-9 MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-justify-xs-space-between"
      >
        <div
          class="MuiGrid-root MuiGrid-item"
        >
          <h4
            class="MuiTypography-root makeStyles-primary-10 MuiTypography-h4"
          />
        </div>
        <div
          class="MuiGrid-root MuiGrid-item"
        >
          <h4
            class="MuiTypography-root makeStyles-secondary-11 MuiTypography-h4"
          />
        </div>
        <div
          class="MuiGrid-root MuiGrid-item"
        >
          <h4
            class="MuiTypography-root makeStyles-secondary-11 MuiTypography-h4"
          />
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root makeStyles-iconsCardSmall-3 MuiGrid-item"
    >
      <div
        class="MuiBox-root MuiBox-root-14 makeStyles-boxStyle-2"
      >
        <div>
          <div>
             
          </div>
          <div
            class="makeStyles-flexDisplay-16"
          >
            <div>
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root makeStyles-root-18"
                data-testid="bus"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M4 16c0 .88.39 1.67 1 2.22V20c0 .55.45 1 1 1h1c.55 0 1-.45 1-1v-1h8v1c0 .55.45 1 1 1h1c.55 0 1-.45 1-1v-1.78c.61-.55 1-1.34 1-2.22V6c0-3.5-3.58-4-8-4s-8 .5-8 4v10zm3.5 1c-.83 0-1.5-.67-1.5-1.5S6.67 14 7.5 14s1.5.67 1.5 1.5S8.33 17 7.5 17zm9 0c-.83 0-1.5-.67-1.5-1.5s.67-1.5 1.5-1.5 1.5.67 1.5 1.5-.67 1.5-1.5 1.5zm1.5-6H6V6h12v5z"
                />
              </svg>
            </div>
            <div>
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root makeStyles-root-18"
                data-testid="metro"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <circle
                  cx="8.5"
                  cy="14.5"
                  r="1.5"
                />
                <circle
                  cx="15.5"
                  cy="14.5"
                  r="1.5"
                />
                <path
                  d="M12 2c-4 0-8 .5-8 4v9.5C4 17.43 5.57 19 7.5 19L6 20.5v.5h2l2-2h4l2 2h2v-.5L16.5 19c1.93 0 3.5-1.57 3.5-3.5V6c0-3.5-4-4-8-4zm0 2c3.51 0 4.96.48 5.57 1H6.43c.61-.52 2.06-1 5.57-1zM6 7h5v3H6V7zm12 8.5c0 .83-.67 1.5-1.5 1.5h-9c-.83 0-1.5-.67-1.5-1.5V12h12v3.5zm0-5.5h-5V7h5v3z"
                />
              </svg>
            </div>
            <div>
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root makeStyles-root-18"
                data-testid="car"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M18.92 5.01C18.72 4.42 18.16 4 17.5 4h-11c-.66 0-1.21.42-1.42 1.01L3 11v8c0 .55.45 1 1 1h1c.55 0 1-.45 1-1v-1h12v1c0 .55.45 1 1 1h1c.55 0 1-.45 1-1v-8l-2.08-5.99zM6.85 6h10.29l1.04 3H5.81l1.04-3zM19 16H5v-4.66l.12-.34h13.77l.11.34V16z"
                />
                <circle
                  cx="7.5"
                  cy="13.5"
                  r="1.5"
                />
                <circle
                  cx="16.5"
                  cy="13.5"
                  r="1.5"
                />
              </svg>
            </div>
            <div>
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root makeStyles-root-18"
                data-testid="bike"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12.72 11l-2 2h-.77l-.25-.69c-.18-.48-.42-.92-.72-1.31h3.74m2.69-6H11v2h3.59l2 2H5c-2.8 0-5 2.2-5 5s2.2 5 5 5c2.46 0 4.45-1.69 4.9-4h1.65l2.77-2.77c-.21.54-.32 1.14-.32 1.77 0 2.8 2.2 5 5 5s5-2.2 5-5c0-2.65-1.97-4.77-4.56-4.97L15.41 5zM19 17c-1.66 0-3-1.34-3-3s1.34-3 3-3 3 1.34 3 3-1.34 3-3 3zM5 17c-1.63 0-3-1.37-3-3s1.37-3 3-3c1.28 0 2.4.85 2.82 2H5v2h2.82C7.4 16.15 6.28 17 5 17z"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
